pkg/client/unversioned/request.go

```diff
@@ -425,7 +425,36 @@ func (r *Request) VersionedParams(obj runtime.Object, convertor runtime.ObjectCo
 	}
 	for k, v := range params {
 		for _, vv := range v {
-			r.setParam(k, vv)
+			// TODO: Move it to setParam method, once we get rid of
+			// FieldSelectorParam & LabelSelectorParam methods.
+			value := vv
+			if k == unversioned.LabelSelectorQueryParam(r.apiVersion) && value == "" {
+				// Don't set an empty selector for backward compatibility.
```

Comment:
We just didn't set the corresponding url param it that wasn't set. So I didn't want to explicitly set fieldSelector=&labelSelector= in every request.